app/src/main/java/org/mozilla/fenix/compose/tabstray/ListTabItem.kt

```diff
@@ -0,0 +1,170 @@
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.browser.tabstray.TabsTray
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.TabImage
+import org.mozilla.fenix.compose.TabTitle
+import org.mozilla.fenix.compose.TabUrl
+import org.mozilla.fenix.ext.components
+
+@Composable
+fun ListTabItem(
+    tab: TabSessionState,
+    isSelected: Boolean = false,
+    multiSelectionEnabled: Boolean = false,
+    multiSelectionSelected: Boolean = false,
+    delegate: TabsTray.Delegate? = null
+) {
+    val contentBackgroundColor = if (isSelected) {
+        R.color.tab_tray_item_selected_background_normal_theme
+    } else {
+        R.color.tab_tray_item_background_normal_theme
+    }
+    val descriptionWeight = if (multiSelectionEnabled) 4f else 3f
+
+    Row(modifier = Modifier
+        .fillMaxWidth()
+        .height(dimensionResource(id = R.dimen.tab_tray_tab_item_height))
+        .background(colorResource(id = contentBackgroundColor))) {
+
+        Thumbnail(tab, multiSelectionSelected)
+
+        DescriptionSection(descriptionWeight, tab)
+
+        CloseIcon(
+            tab = tab,
+            visible = !multiSelectionEnabled,
+            delegate = delegate
+        )
+    }
+}
+
+@Composable
+private fun RowScope.Thumbnail(tab: TabSessionState, isSelected:Boolean) {
+    Box(
+        modifier = Modifier.Companion
+            .weight(2f)
+            .padding(start = 16.dp, end = 12.dp, top = 8.dp, bottom = 8.dp)
+    ) {
+        TabImage(
+            tab = tab,
+            modifier = Modifier
+                .size(
+                    dimensionResource(id = R.dimen.tab_tray_list_item_thumbnail_width),
+                    dimensionResource(id = R.dimen.tab_tray_list_item_thumbnail_height)
+                ),
+            thumbnailDescription = stringResource(id = R.string.mozac_browser_tabstray_open_tab),
+            contentScale = ContentScale.FillWidth,
+            alignment = Alignment.TopCenter
+        )
+
+        if (isSelected) {
+            Card(
+                modifier = Modifier
+                    .size(40.dp)
+                    .align(Alignment.Center),
+                shape = CircleShape,
+                backgroundColor = colorResource(id = R.color.accent_normal_theme),
+            ) {
+                Icon(
+                    painter = painterResource(id = R.drawable.mozac_ic_check),
+                    modifier = Modifier
+                        .matchParentSize()
+                        .padding(8.dp),
+                    contentDescription = null,
+                    tint = colorResource(id = R.color.mozac_ui_icons_fill)
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun RowScope.CloseIcon(
+    tab: TabSessionState,
+    visible: Boolean,
+    delegate: TabsTray.Delegate?
```

Comment:
Rather than doing this, you can define a lambda function such as `onCloseClick`.